File: Tyche.Android/bet/src/main/java/com/felipearpa/tyche/bet/domain/Bet.kt
package com.felipearpa.tyche.bet.domain

import com.felipearpa.tyche.core.type.BetScore
import com.felipearpa.tyche.core.type.Ulid

data class Bet(
    val poolId: Ulid,
    val gamblerId: Ulid,
    val matchId: Ulid,
    val homeTeamBet: BetScore,
    val awayTeamBet: BetScore
)

File: Amazon.Lambda/Felipearpa.Tyche.Pool/src/Application/Bet.fs
namespace Felipearpa.Tyche.Pool.Application

open Felipearpa.Tyche.Pool.Domain
open Felipearpa.Tyche.Pool.Type
open Felipearpa.Type

type Bet(poolGamblerBetRepository: IPoolGamblerBetRepository) =

    member this.ExecuteAsync(poolId: Ulid, gamblerId: Ulid, matchId: Ulid, betScore: TeamScore<BetScore>) =
        poolGamblerBetRepository.BetAsync(poolId, gamblerId, matchId, betScore)
